Add navi

SteveLoveWork opened the pull request:
- The instruction will update by triggering the +1 on the instuction list for now as the routing does not work.
- However the framework has been setup already for when a new position of the user is within 5.0m of the node it will trigger the text to speech

jzpnhlim committed:
add routing

jzpnhlim committed:
add function that find long/lat by room

mostafaayesh committed:
replace http with dio

mostafaayesh committed:
Fix typo in nominatim.dart

SteveLoveWork committed:
Added Text to Speech

- The instruction will update by triggering the +1 on the instuction list for now as the routing does not work.
- However the framework has been setup already for when a new position of the user is within 5.0m of the node it will trigger the text to speech

SteveLoveWork committed:
Revert "Added Text to Speech"

This reverts commit dfe4183654f27ad66675aa9082dac0e2c0548499.

SteveLoveWork committed:
Revert "Revert "Added Text to Speech""

This reverts commit c3a546a0e569106f4ca57ed3d0f0a953fcfe9db1.

SteveLoveWork committed:
Revert "Revert "Revert "Added Text to Speech"""

This reverts commit b3dbfb90d35906bec563001ac145e2b037d581c7.

SteveLoveWork committed:
Added Text to Speech

- The instruction will update by triggering the +1 on the instuction list for now as the routing does not work.
- However the framework has been setup already for when a new position of the user is within 5.0m of the node it will trigger the text to speech

MostafaAyesh commented:
Going over the code, I see that to update the distance you have to tap the button which isn't the expected functionality. The functionality should be on a user coordinate update a check is made calculating the distance between the user's newly acquired coordinates (x,y) and the coordinates (x,y) of the next instruction. 

To achieve the functionality, some of the changes required are:
- Create a user model (extends ChangeNotifier) to track the user's position
- Create a listener for BLE notifications from the tag to update the user's model
- Create a listener in the navigation activity that listens for user model updates and makes a decision to update the widgets and trigger the TTS when the time is right (using distance calculations or timers)

SteveLoveWork committed:
Update pubspec.yaml

Add dependency for text to speech

SteveLoveWork committed:
Update Instruction List

The current Instruction and instruction list will update as the +1 button is pressed.
- The +1 button is just a back door to manually update the instruction list since tracking has not been implemented

jzpnhlim committed:
improves valhalla module

imporved valhalla module

fetch route function now returns the full json response.

add getShape() funciton to get polyline

add getInstruction() function to get the instruction set

add test cases for valhalla module

jzpnhlim committed:
correct class type mismatch

SteveLoveWork committed:
Merge branch 'add_routing' into add_navi

jzpnhlim committed:
get all access to nominatim output

added seperate getGlobalCoordinates(NominatimOutput) function

tested

jzpnhlim committed:
fixed type conflict and add funciton convert val instruction to instruciton type

tested

SteveLoveWork committed:
Merge branch 'add_routing' into add_navi
